controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -376,7 +395,8 @@ var configMapPredicates = predicate.Funcs{
 // a workaround for 2.5 due to odh-model-controller serivceaccount keeps updates with label.
 var saPredicates = predicate.Funcs{
 	UpdateFunc: func(e event.UpdateEvent) bool {
-		if e.ObjectNew.GetName() == "odh-model-controller" && (e.ObjectNew.GetNamespace() == "redhat-ods-applications" || e.ObjectNew.GetNamespace() == "opendatahub") {
+		namespace := e.ObjectNew.GetNamespace()
+		if e.ObjectNew.GetName() == "odh-model-controller" && (namespace == cluster.RHOAIApplicationNamespace || namespace == cluster.ODHApplicationNamespace) {
```

Comment:
sorry, after a second review, I think my comment was incorrect in the assumption that the namespace were the one where the operator runs. However, I think it would be better to take the namespace from the DSCI instead of having some hardcoded values.

Reply:
yeah, i have reverted the change and can have another PR to work on the namespace part